packages/smart-contracts/src/contracts/ERC20SwapToPay.sol

```diff
@@ -118,24 +150,51 @@ contract ERC20SwapToPay is Ownable {
     );
 
     // Give the change back to the payer, in both currencies (only spent token should remain)
-
     if (spentToken.balanceOf(address(this)) > 0) {
       spentToken.safeTransfer(msg.sender, spentToken.balanceOf(address(this)));
     }
+
     if (requestedToken.balanceOf(address(this)) > 0) {
       requestedToken.safeTransfer(msg.sender, requestedToken.balanceOf(address(this)));
     }
   }
 
+  /**
+   * Internal function to calculate the amount of request fee to pay with a transaction.
+   * @param _amount The Wei amount to calculate the fee from.
+   * @dev The _amount has to be nominated in Wei.
+   * @dev returns the _feeAmount in Wei. 
+   */
+  function _calculateSwapFee(uint256 _amount) internal view returns (uint256 _swapFeeAmount) {
+    _swapFeeAmount = (_amount.mul(currentSwapFee)).div(1000);
```

Comment:
why 1000 🤔 ?

Reply:
The reason was due to the `swapFee` being 0.5%.
In `setSwapFee(uint256 _newSwapFee)` the minimum input value would be one digit. 

For example: let's get a 0.5% swap fee of 1000.  
If we have the `_newSwapFee` set to `5`. 
then `1000.mul(5).div(1000)` should return `5` if I'm not mistaken?

Reply:
I thought it could be something like this, but then it should be very clearly documented in the contract
maybe even increase to 2 decimals, to have more control over the rate, like `1.25` ? (not sure about this)

Reply:
Yes, I did write it as a `@dev` comment and gave three examples. 
But I'm always open to suggestions to better documenting it.

Reply:
fixed